components/omega/src/ocn/OceanState.cpp

```diff
@@ -428,6 +525,32 @@ OceanState *OceanState::get(const std::string Name ///< [in] Name of state
    }
 } // end get state
 
+//------------------------------------------------------------------------------
+// Get time index from time level
+// TimeLevel == [1:new, 0:current, -1:previous, -2:two times ago, ...]
+I4 OceanState::getTimeIndex(I4 &TimeIndex, const I4 TimeLevel) const {
+
+   // Handle single time level case (no new time level)
+   I4 TimeLevelAdj;
+   if (NTimeLevels == 1) {
+      TimeLevelAdj = TimeLevel;
+   } else {
+      TimeLevelAdj = TimeLevel - 1;
+   }
+
+   // Check if time level is valid
+   if (TimeLevelAdj > 0 || (TimeLevelAdj + NTimeLevels) <= 0) {
+      LOG_ERROR("OceanState: Time level {} is out of range for NTimeLevels {}",
+                TimeLevel, NTimeLevels);
+      return -1;
+   }
+
+   TimeIndex = (TimeLevelAdj + CurTimeIndex + NTimeLevels) % NTimeLevels;
+
+   return 0;
+
+} // end get time index
+
```

Comment:
The Tracers module has been updated to strictly follow the time level convention in the Ocean state, eliminating the need to adjust the time level in this function.

```suggestion
//------------------------------------------------------------------------------
// Get time index from time level
// TimeLevel == [1:new, 0:current, -1:previous, -2:two times ago, ...]
I4 OceanState::getTimeIndex(I4 &TimeIndex, const I4 TimeLevel) const {

   // Check if time level is valid
   if (NTimeLevels > 1 && (TimeLevel > 1 || (TimeLevel + NTimeLevels) <= 1)) {
      LOG_ERROR("OceanState: Time level {} is out of range for NTimeLevels {}",
                TimeLevel, NTimeLevels);
      return -1;
   }

   TimeIndex = (TimeLevel + CurTimeIndex + NTimeLevels) % NTimeLevels;

   return 0;

} // end get time index
```

Reply:
Thanks for the suggestion, @grnydawn.